File: TWO-SUM/ts.go
package main

import (
	"fmt"
	"math"
)

func twoSum(nums []int, target int) []int {
	m:= make(map[int]int)
	for i, val:= range nums{
		dif := int(math.Abs(float64(target - val)))
		if index,ok := m[dif]; ok {
			return []int{index, i}
		}
		m[val] = i
	}
	return nil
}

func main(){
	fmt.Println(twoSum([]int{2,7,11,15}, 9))
}

File: TWO-SUM/ts.py
class Solution:
    def twoSum(self, nums, target):
        hm = {}
        for i, num in enumerate(nums):
            dif = target - num
            if dif in hm:
                return [hm[dif], i]
            hm[num] = i
        return {}

class main:
    obj = Solution()
    print(obj.twoSum([2,7,11,15],9))

if __name__ == "__main__":
    main()